hr_timesheet_sheet_no_negative/models/account_analytic_line.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    def _negative_unit_amount(self):
+        if not self.company_id:
+            user_company = self.env.user.company_id
+            company_id = self._context.get('company_id') or user_company.id
+            company = self.env['res.company'].browse(company_id)
```

Comment:
`user_company` is already a record so, in case no `company_id` in context, you are converting it to an `id` and then back to a record. I think this block could be written in a smarter way.
Also `company_id` is a related field and it's required in `account.analytic.account`, so I think this block of code is not necessary at all.